File: cinema.log.server/Models/Entities/UserFilmRating.cs
namespace cinema.log.server.Models.Entities;

public class UserFilmRating
{
    public Guid UserFilmRatingId { get; set; }
    public User User { get; set; }
    public required Guid UserId { get; set; }
    public Film Film { get; set; }
    public required Guid FilmId { get; set; }
    public double EloRating { get; set; }
    public int NumberOfComparisons { get; set; }
    public double KConstantValue { get; set; } = 40;
    public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
    public float InitialRating { get; set; }
}

File: cinema.log.server.golang/internal/domain/user_film_rating.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type UserFilmRating struct {
	ID                  uuid.UUID `json:"id"`
	UserId              uuid.UUID `json:"userId"`
	FilmId              uuid.UUID `json:"filmId"`
	EloRating           float64   `json:"eloRating"`
	NumberOfComparisons int       `json:"numberOfComparisons"`
	LastUpdated         time.Time `json:"lastUpdated"`
	InitialRating       float32   `json:"initialRating"`
	KConstantValue      float64   `json:"kConstantValue"`
}

type UserFilmRatingDetail struct {
	Rating    UserFilmRating `json:"rating"`
	FilmTitle string         `json:"filmTitle"`
	FilmReleaseYear string   `json:"filmReleaseYear"`
	FilmPosterURL string     `json:"filmPosterUrl"`
}

type ComparisonPair struct {
    FilmA UserFilmRating
    FilmB UserFilmRating
}